controllers/toolchainconfig/toolchainconfig_controller_test.go

```diff
@@ -6,103 +6,173 @@ import (
 	"testing"
 
 	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
-	. "github.com/codeready-toolchain/toolchain-common/pkg/test"
+	. "github.com/codeready-toolchain/host-operator/test"
+
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	testconfig "github.com/codeready-toolchain/toolchain-common/pkg/test/config"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-func TestReconcileWhenToolchainConfigIsAvailable(t *testing.T) {
+func TestReconcile(t *testing.T) {
 	// given
-	config := newToolchainConfigWithReset(t, AutomaticApproval().MaxUsersNumber(123, PerMemberCluster("member1", 321)))
-	cl := NewFakeClient(t, config)
-	controller := Reconciler{
-		Client: cl,
-		Log:    ctrl.Log.WithName("controllers").WithName("ToolchainConfig"),
-	}
-
-	// when
-	_, err := controller.Reconcile(newRequest())
-
-	// then
-	require.NoError(t, err)
-	actual, err := GetConfig(NewFakeClient(t), HostOperatorNs)
-	require.NoError(t, err)
-	assert.False(t, actual.AutomaticApproval().IsEnabled())
-	assert.Equal(t, 123, actual.AutomaticApproval().MaxNumberOfUsersOverall())
-	assert.Equal(t, config.Spec.Host.AutomaticApproval.MaxNumberOfUsers.SpecificPerMemberCluster, actual.AutomaticApproval().MaxNumberOfUsersSpecificPerMemberCluster())
-
-	t.Run("update with new version", func(t *testing.T) {
-		// given
-		threshold := 100
-		config.Spec.Host.AutomaticApproval.ResourceCapacityThreshold.DefaultThreshold = &threshold
-		err := cl.Update(context.TODO(), config)
-		require.NoError(t, err)
-
-		// when
-		_, err = controller.Reconcile(newRequest())
-
-		// then
-		require.NoError(t, err)
-		actual, err := GetConfig(NewFakeClient(t), HostOperatorNs)
-		require.NoError(t, err)
-		assert.False(t, actual.AutomaticApproval().IsEnabled())
-		assert.Equal(t, 123, actual.AutomaticApproval().MaxNumberOfUsersOverall())
-		assert.Equal(t, config.Spec.Host.AutomaticApproval.MaxNumberOfUsers.SpecificPerMemberCluster, actual.AutomaticApproval().MaxNumberOfUsersSpecificPerMemberCluster())
-		assert.Equal(t, 100, actual.AutomaticApproval().ResourceCapacityThresholdDefault())
+	defaultMemberConfig := testconfig.NewMemberOperatorConfig(testconfig.MemberStatus().RefreshPeriod("5s"))
+	specificMemberConfig := testconfig.NewMemberOperatorConfig(testconfig.MemberStatus().RefreshPeriod("10s"))
+
+	t.Run("success", func(t *testing.T) {
```

Comment:
Could you please verify in any of the tests that the member configs were really updated? In other words, that the synchronization logic was triggered?

Reply:
Done with efe10d48c01cf0288d61cf0898f5983448f33ca3 and da1c26c3963382219fa07206e74cccd1949f4ac7